### DISCUSSION TITLE
Control has already been added to another page: {}".format(ctrl)

### DISCUSSION THREAD

**Discussion by hoangde2811** [flet-dev/flet#1192] [Q&A]
Dear developer!!
I create a progress to wait for data to load, for that I used dialog and have await page.update_async , when opening a browser tab there is no problem but opening another browser tab gives an error as attached picture. 
Please help me so I can run multiple tabs. 
Sincerely thank you.
![image](https://user-images.githubusercontent.com/87251371/226081457-51a79651-f463-4bcb-bc86-77e86108cdc7.png)

**Comment by FeodorFitsner** [comment]
Please provide sample code reproducing the issue.

**Comment by hoangde2811** [comment]
ERROR:asyncio:Task exception was never retrieved
future: <Task finished coro=<__connect_internal_async.<locals>.on_event() done, defined at C:\NewFlet\lib\site-packages\flet\flet.py:361> exception=Exception("Control has already been added to another page: alertdialog {'open': True, 'modal': False, 'actionsalignment': 'end'}")>
Traceback (most recent call last):
  File "C:\NewFlet\lib\site-packages\flet\flet.py", line 364, in on_event
    Event(e.eventTarget, e.eventName, e.eventData)
  File "C:\NewFlet\lib\site-packages\flet_core\page.py", line 456, in on_event_async
    await handler(ce)
  File "D:\backup\Python\NewFlet\views\login.py", line 65, in login
    await ProgressBar(page).show()
  File "D:\backup\Python\NewFlet\Utils\progress_bar.py", line 23, in show
    await self.page.update_async()
  File "C:\NewFlet\lib\site-packages\flet_core\page.py", line 264, in update_async
    r = await self.__update_async(self)
  File "C:\NewFlet\lib\site-packages\flet_core\page.py", line 368, in __update_async
    self.__validate_controls_page(added_controls)
  File "C:\NewFlet\lib\site-packages\flet_core\page.py", line 395, in __validate_controls_page
    "Control has already been added to another page: {}".format(ctrl)
Exception: Control has already been added to another page: alertdialog {'open': True, 'modal': False, 'actionsalignment': 'end'}

Above is the error code when opening 2 browser tabs. 
(https://user-images.githubusercontent.com/87251371/226088895-0030186d-c778-4501-b6ca-c57e93c2f873.png) and here is my progress dialog file.
(https://user-images.githubusercontent.com/87251371/226089756-dffb1db1-2df3-40f8-855f-52231edf759b.png) and here is my main file
Sincerely thank you.
